packages/browser-wallet/src/background/index.ts

```diff
@@ -303,13 +314,48 @@ bgMessageHandler.handleMessage(
     getMostRecentlySelectedAccountHandler
 );
 
+const withPromptStart: RunCondition<MessageStatusWrapper<string | undefined>> = async () => {
+    const isPromptOpen = await sessionOpenPrompt.get();
+    if (isPromptOpen) {
+        return { run: false, response: { success: false, message: 'Another prompt is already open' } };
+    }
+    sessionOpenPrompt.set(true);
+    return { run: true };
+};
+
+function withPromptEnd() {
+    sessionOpenPrompt.set(false);
+}
```

Comment:
I could't do the following, but what is there any way this doesn't get triggered? Then the user will be in a state where no new requests can be opened.

Alternatively, we have a `background/window-management.ts@testPopupOpen`, which we can use, that doesn't depend on setting stuff on session storage, which could potentially block users.

Reply:
I don't believe this can not be triggered (without closing the browser).

It is an interesting idea, but it would also block if you any window open. (the popup itself, or recovery results, etc.)
I'm don't know whether we want to do that.

Reply:
Yes, you're right.. don't know what I was thinking. Maybe instead of doing sessionstorage, could we do something similar to `testPoopupOpen` where we ping the prompt page to test? I think it's the safer approach of the two.